### PR TITLE
ROX-17671:  prune log imbues with SQL

### PR DESCRIPTION
**Pull request by dashrews78**
## Description

Update of the pruning of log imbues to use SQL instead of walking all the data.  This is much simpler, we just had to promote the time field to be a column in postgres so that we could use it.  I did not want to write a migration for this as it is only used for pruning, but since we keep these log imbues for a week, the safest path was to write a migration to ensure that the time field was populated immediately after an upgrade.

(Though if I was smart I would have snuck the schema change into 4.2 and then the prune in 4.3 and then I wouldn't have needed the migration, but alas....)

## Checklist
- [x] Investigated and inspected CI test results
- [x] Unit test and regression tests added
~- [ ] Evaluated and added CHANGELOG entry if required~
~- [ ] Determined and documented upgrade steps~
~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs]~(https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Existing pruning tests covered testing the new pruning function.
For the migration I wrote a migration test which:
- wrote data to the old schema 
- updated the schema
- ensured the time column was NULL
- ran the migration
- ensured the time column was NOT NULL

Additionally, I fired up a 4.1.3 cluster:
- wrote some log imbues
- checked the DB to ensure no time column was there
- updated to image from this PR
- checked the DB to ensure time column was populated
- rolled back to 4.1.3
- added some log imbues
- checked the DB to ensure time column was populated for entries before the rollback and null for entries after the rollback
- re-upgraded to image from this PR
- checked the DB to ensure time column was populated for all entries.

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: migrator/migrations/m_188_to_m_189_logimbues_add_time_column/test/schema/convert_log_imbues.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Code originally generated by pg-bindings generator.
```
Comment (c-du) [inline migrator/migrations/m_188_to_m_189_logimbues_add_time_column/test/schema/convert_log_imbues.go:1]: Is this the same as migrator/migrations/m_188_to_m_189_logimbues_add_time_column/schema/convert_log_imbues.go can you reuse it?

**Review by c-du — APPROVED** [review]

File: migrator/migrations/m_188_to_m_189_logimbues_add_time_column/test/schema/convert_log_imbues.go (hunk 1 above)
Comment (dashrews78) [reply migrator/migrations/m_188_to_m_189_logimbues_add_time_column/test/schema/convert_log_imbues.go:1]: @c-du No, this one is to populate the old schema for the tests.  This one does not translate the time whereas the other does convert the time field for the new schema.  I'm sure I could probably figure out a way to do something with this but for one time test code, does not seem worth it.

Comment (dashrews78) [reply migrator/migrations/m_188_to_m_189_logimbues_add_time_column/test/schema/convert_log_imbues.go:1]: @c-du No, this one is to populate the old schema for the tests.  This one does not translate the time whereas the other does convert the time field for the new schema.  I'm sure I could probably figure out a way to do something with this but for one time test code, does not seem worth it.

**Pull request merged**